src/envs/behavior_options.py

```diff
@@ -0,0 +1,976 @@
+    done."""
+    ids = []
+    for obj in env.scene.get_objects():
+        if isinstance(obj, URDFObject):
+            # We want to exclude the floor since we're always floating and
+            # will never practically collide with it, but if we include it
+            # in collision checking, we always seem to collide.
+            if obj.name != "floors":
+                ids.extend(obj.body_ids)
+
+    if include_self:
+        ids.append(env.robots[0].parts["left_hand"].get_body_id())
+        ids.append(env.robots[0].parts["body"].get_body_id())
+        ids.append(env.robots[0].parts["eye"].get_body_id())
+        if not grasping_with_right:
+            ids.append(env.robots[0].parts["right_hand"].get_body_id())
+
+    return ids
+
+
+def detect_collision(bodyA: int, object_in_hand: int = None) -> bool:
+    """Detects collisions between objects in the scene (except for the object
+    in the robot's hand)"""
+    collision = False
+    for body_id in range(p.getNumBodies()):
+        if body_id in [bodyA, object_in_hand]:
+            continue
+        closest_points = p.getClosestPoints(bodyA, body_id, distance=0.01)
+        if len(closest_points) > 0:
+            collision = True
+            break
+    return collision
+
+
+# NOTE: we need a type:ignore because we can't specify the type of robot because
+# it would be dependent on an optional igibson import which won't get imported
+# on CI
+def detect_robot_collision(robot) -> bool:  # type: ignore
+    """Function to detect whether the robot is currently colliding with any
+    object in the scene."""
+    object_in_hand = robot.parts["right_hand"].object_in_hand
+    return (detect_collision(robot.parts["body"].body_id)
+            or detect_collision(robot.parts["left_hand"].body_id)
+            or detect_collision(robot.parts["right_hand"].body_id,
+                                object_in_hand))
+
+
+# NOTE: we need a type:ignore because we can't specify the type of env because
+# it would be a igibson.envs.behavior_env.BehaviorEnv, which is an optional
+# import and won't get imported on CI
+def reset_and_release_hand(env) -> None:  # type: ignore
+    """Resets the state of the right hand."""
+    env.robots[0].set_position_orientation(env.robots[0].get_position(),
+                                           env.robots[0].get_orientation())
+    for _ in range(50):
+        env.robots[0].parts["right_hand"].set_close_fraction(0)
+        env.robots[0].parts["right_hand"].trigger_fraction = 0
+        p.stepSimulation()
+
+
+# NOTE: we need a type:ignore because we can't specify the type of env because
+# it would be a igibson.envs.behavior_env.BehaviorEnv, which is an optional
+# import and won't get imported on CI
+def get_delta_low_level_base_action(  # type: ignore
+    env,
+    original_orientation: Tuple,
+    old_xytheta: Array,
+    new_xytheta: Array,
+) -> Array:
+    """Get low-level actions from base movement plan."""
```

Comment:
this docstring could be improved -- it's unclear what "base movement plan" means